### ISSUE TITLE
Ac unit not responding to "status cmd" 

### ISSUE THREAD

**Issue by MotiGross** [tomikaa87/gree-remote#47]
Hello all,
many thanks for this great info ! i followed the instructions & open source and developed everything from scratch- AES128 encrypt & Decrypt, BASE64 encrypt, (the BASE64 decrypt was part of exiting lib) on a TwinCAT3.1 PLC.
i was able to "search" and "bind' and even got response from the AC unit with it's special key,
now, I'm trying to read AC unit statuses, but got zero response, 
i think the status cmd i'm sending is wrong,
i used an online tool to double check the "Pack", from some reason i'm not sure why (yet) it's different than what my TwinCAT code is generating, but i send "status cmd" with both Packs ,and AC not responding to any of them.

the only thing cross my mind: is there any meaning for spacing? which i might used differently 
here is how my Json looks like:
{"cols":["Pow","Mod","SetTem","WdSpd","Air","Blo","Health","SwhSlp","Lig","SwingLfRig","SwUpDn","Quiet","Tur","StHt","TemUn","HeatCoolType","TemRec","SvSt"],"mac":"f4911e312b48","t": "status"}

2nd thing that cross my mind : does status cmd is encrypted with the generic key or the unique key ?
tnx! Moti

**Comment by tomikaa87** [comment]
Hi @MotiGross,

Did you try to request just one basic variable (e.g. "Pow") to verifiy if it works? I can imagine that the device rejects the packet due to an unknown parameter.

Status packs must be encrypted with the device key. A rule of thumb is every pack you use after binding the device must be encrypted with the specific key. The generic key is only used to find out the specific device key.

**Comment by MotiGross** [comment]
Hi,
tnx for the feedback, 
regarding the Keys usages, that was quite clear, but since stuff didn't come right, than i suspect i'm might wrong,
any way, i found that root cause in my code
1. i missed the " " at the IP string in my pack before encrypting
2. i found some bugs the AES128 or BASE64 (no solving them yet but working on that ) .
now when generating the pack with online tool the AC unit respond well, so i need to fix my code.

tnx again,

**Comment by MotiGross** [comment]
ok, something weird.... i'll appreciate some gaudiness on this issue: 
i'm sending "get status" : here is my full command as created with my code:
'{"cid":"app","i":0,"pack":"MrFipIy7eUEogtYAq3f4DJutPHf9oxcngHaJyXX9ekJPscYAwLAspTfCUCooDeaZVMW+hu7ckvLmYgWPH0qskBWq1Vb8qUnP57j+d/29LuY1Gs9n0ZCmtKf63VssPA8RyR1BTel+pWnlZiyXXeylgZBvakstAsaq/2zLUUfhHQqcTCXOJO9T7O1M96m8nId/D/OEmd9wr4RxJCqvxxv41Bw3QuGCm664yBAtT8C2B2XQqMf4kATMe1zVw5f8eZe9","t":"pack","tcid":"f4911e312b48","uid":0}'
--> the Pack is ending with Ze9. 

the pack is 192chars before starting with AES128 encrypt, while comparing every single char with online tool ,it's 100% compatible!
than, it goes to encrypt with BASE64, so, 192/3*4 =  256 and that exactly what my base64 encrypt code is generating
when i'm using a online tool to encrypt the AES128 result, the result to base64 is longer , and those chars are added at the end:
mKQhMjOetsd8nCEvPXLT4A==

with my base64 the AC unit not responding, with the extra chars it respond successfully, 

what am i missing here? 
what are those extra "mKQhMjOetsd8nCEvPXLT4A==" and why the AC unit expecting them to be presented? 

note, that the search command use the exact same methods, and it respond with the special key.
tnx Moti

**Comment by MotiGross** [comment]
ok, problem found with the specification of " PKCS-7 padding" if a string is "round" =filling all blocks, than another block must be added with padding .
works :-)

**Comment by jodewee** [comment]
@MotiGross where did you adjust the code for that one?

**Comment by MotiGross** [comment]
Hi, i'm not using any of the existing code here, my platform (as i explained in the 1st message here) is TwinCAT PLC, and there for the code here not useable for me at any way, 
i do use the code to understand the concept, but as i wrote, my code was created from scratch, even the AES128 encrypt & decrypt and BASE64 encrypt & decrypt.
i'll add here my code when i'll get it stable m for future TwinCAT PLC programmer .

**Comment by tomikaa87** [comment]
Closing because it's a duplicate of https://github.com/tomikaa87/gree-remote/issues/54.
